Lint/formatter

EmielM opened the pull request:
It needs some!

Tried out biome with default settings, but that is rather different from a lot of current code formatting. The big ones:
- Tabs
- Statements end with `;`
- Always braces, so no if/while oneliners
- Prefers double quotes
- Sometimes prefers fat arrow functions over normal ones [not sure why]

To do
- [ ] Around 40 violations left, mainly cases of `any`
- [x] A test now fails (did the `--fix --unsafe`)
- [ ] biome feels pretty unorthodox, perhaps a more conventional combo eslint/prettier fits better
- [ ] Integrate in CI flow? (and `bun test` too)

vanviegen commented:
I agree that some linting is well overdue, and Biome's default style looks fine to me.

The bug that it introduces appears to be due to it having replaced all occurances of  `x instanceof Array` with `Array.isArray(x)`. The latter, however, returns false for arrays wrapped by a Proxy, while the former returns true. So we'd somehow need to instruct Biome not to touch those.

Apart from that, I'd say: let's do this. And indeed, work towards adding it as a CI step eventually.

EmielM commented:
Rebased &
- Updated biome config to disable [useIsArray](https://biomejs.dev/linter/rules/use-is-array/), tests now pass indeed
- Autofix patch
- Some easy fixes

35 violations left, of which 7 in reverseSortedSet, which will probably be solved by the other pull request.

Suggestion to to perhaps merge (squashed?) for now, and resolve the others independently?
